Add Sequelize ORM and Replace SQL Queries

chnthkksn opened the pull request:
In this PR, I have made the following modifications:

1. Added Sequelize as a dependency
2. Configured Sequelize connection
3. Migrated SQL queries to Sequelize
4. Created database models
5. Refactored code for Sequelize usage

chnthkksn committed:
Added Sequelize to the backend
- removed all raw SQL queries and replaced them with Sequelize queries

chnthkksn committed:
Merge branch 'main' into sequelize

chnthkksn commented:
There were some conflicts with the main branch. I have resolved them.

nimi-kaushal commented on server/models/categoryModel.js:
```diff
@@ -0,0 +1,23 @@
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      allowNull: false,
```
remove : allowNull: false,

chnthkksn replied:
removed `allowNull: false` for primary keys in commit 9a6b72d22fc7b1046356ec2c70ed77825b0bb0b4

nimi-kaushal commented on server/models/transactionModel.js:
```diff
@@ -0,0 +1,35 @@
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      allowNull: false,
```
remove    `allowNull: false,`

chnthkksn replied:
removed `allowNull: false` for primary keys in commit 9a6b72d22fc7b1046356ec2c70ed77825b0bb0b4

nimi-kaushal commented on server/routes/budgetRouter.js:
```diff
@@ -1,11 +1,11 @@
-const router = require('express').Router();
```
use single ' quotation intsend of  " double quotation

nimi-kaushal commented on server/models/transactionModel.js:
```diff
@@ -0,0 +1,35 @@
+    },
+  },
+  {
+    tableName: "transactions",
```
```
      sequelize 
      tableName: 'transaction',
      underscored: true
```

chnthkksn replied:
added in commit 9a6b72d22fc7b1046356ec2c70ed77825b0bb0b4

nimi-kaushal commented on server/models/budgetModel.js:
```diff
@@ -0,0 +1,30 @@
+    },
+  },
+  {
+    tableName: "budgets",
```
```
sequelize 
  tableName: 'budget',
      underscored: true
```

chnthkksn replied:
added in commit 9a6b72d22fc7b1046356ec2c70ed77825b0bb0b4

nimi-kaushal commented on server/controllers/budgetController.js:
```diff
@@ -1,74 +1,71 @@
+const Category = require("../models/categoryModel");
 
 module.exports = {
+  getBudget: async (req, res) => {
```
assign to `const BudgetContoller `and export the module ,  then no need to import as seperate methods

chnthkksn replied:
fixed in commit 099eb1dd4c684658b8f51ed33da07bae864438a8

nimi-kaushal commented on server/models/budgetModel.js:
```diff
@@ -0,0 +1,30 @@
+const { DataTypes } = require("sequelize");
+const sequelize = require("../database/sequelize");
+
+const Budget = sequelize.define(
```
extend from model

chnthkksn replied:
removed sequelize.define and added class Model extends in commit 9a6b72d22fc7b1046356ec2c70ed77825b0bb0b4

chnthkksn committed:
Refactored contollers to use inline imports

chnthkksn committed:
Merge branch 'sequelize' of https://github.com/Nimi-NimiX/pet-project-chinthaka into sequelize

chnthkksn committed:
Fixed issues in sequelize
- used class extends Model instead of sequelize.define
- added underscored and sequelize to models
- removed allowNull from primary key in models
- moved associations to models from connection.js
